src/main/java/org/mqttbee/mqtt/handler/MqttChannelInitializer.java

```diff
@@ -54,9 +57,21 @@
 
     @Override
     protected void initChannel(final SocketChannel channel) {
-        final ChannelComponent channelComponent = ChannelComponent.create(channel, clientData);
+        init(channel);
+        addMqttHandlers();
```

Comment:
Pass the channel to the method, so the pipeline attribute can be removed from the class

Reply:
done